src/Umbraco.Web/PublishedContentQuery.cs

```diff
@@ -224,14 +225,36 @@ private IPublishedContent TypedDocumentById(int id, ContextualPublishedCache cac
         {
             var doc = cache.GetById(id);
             return doc;
-        }
+        }
+
+        private static ConcurrentDictionary<Guid, int> _guidToIntLoopkup;

```

Comment:
Why not initialize it straight away instead of incurring the cost of a null check every time `TypedDocumentById` is called? An empty dictionary costs so little plus the check is not thread safe.

`private static readonly ConcurrentDictionary<Guid, int> GuidToIntLoopkup =  new ConcurrentDictionary<Guid, int>();`

Reply:
I guess I was trying to initialising it once `TypedDocumentById(Guid)` is called - as opposed to whenever `PublishedContentQuery` is instantiated. _e.g. if you don't that method, then it wouldn't be created?_

Reply:
Init cost is very little compared to the null check hit. As soon as the class is referenced anywhere the reference will be created.